filebeat/scripts/mage/generate/module.go

```diff
@@ -15,38 +15,33 @@
 // specific language governing permissions and limitations
 // under the License.
 
-package main
+package generate
 
 import (
-	"flag"
 	"fmt"
 	"os"
 
-	"github.com/elastic/beats/filebeat/generator/fields"
+	devtools "github.com/elastic/beats/dev-tools/mage"
+	genmod "github.com/elastic/beats/filebeat/generator/module"
 )
 
-func main() {
-	module := flag.String("module", "", "Name of the module")
-	fileset := flag.String("fileset", "", "Name of the fileset")
-	beatsPath := flag.String("beats_path", ".", "Path to elastic/beats")
-	noDoc := flag.Bool("nodoc", false, "Generate documentation for fields")
-	flag.Parse()
-
-	if *module == "" {
-		fmt.Println("Missing parameter: module")
-		os.Exit(1)
+// Module creates a new Filebeat module.
+// Use MODULE=module to specify the name of the new module
+func Module() error {
+	targetModule := os.Getenv("MODULE")
+	if targetModule == "" {
+		return fmt.Errorf("You must specify the module: MODULE=name mage generate:module")
```

Comment:
I like to always start error strings with lowercase chars. https://github.com/golang/go/wiki/CodeReviewComments#error-strings